src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -479,3 +489,41 @@ between open close p = tok open *> p <* tok close
 -- `empty` /= `pure mempty` for `Parser`.
 optempty :: Monoid a => Parser a -> Parser a
 optempty = option mempty
+
+-- | Parses strings delimited by triple quotes. 
+-- http://spec.graphql.org/June2018/#sec-String-Value
+blockString :: Parser Text
+blockString = do
+  _ <- triplequotes <?> "opening triple quotes"
+  str <- T.lines . T.pack <$> AT.manyTill AT.anyChar triplequotes <?> "the body of a triple quoted string"
+  let !cleanLines = dropEmptyLines str
+  let smallest = foldr selectSmallest maxBound cleanLines -- counts indentation and selects the smallest one from all lines
+  let fixedLines = foldr (fixIndentation smallest) "" cleanLines -- removes the common indentation from all lines.
+  return (T.strip . toStrict . TB.toLazyText $ fixedLines)
+ where
+  -- used to remove the common indentation from each line.
+  fixIndentation :: Int -> Text -> TB.Builder -> TB.Builder 
+  fixIndentation smallest a acc =
+    let new = TB.fromText $ T.dropWhileEnd (== ' ') (T.drop smallest a)
+    in new <> "\n" <> acc
+  -- used to accumulate the smallest indentation
+  selectSmallest :: Text -> Int -> Int
+  selectSmallest t acc =
+    let new = countIndentation t
+     in if (not $ onlySpaces t) && new < acc
+       then new
+       else acc
```

Comment:
Instead of folding over the lines like this, why don't we just take a `min` over the `countIndentation`s?

Reply:
On a first attempt it did not work. I'll try again tomorrow.

Reply:
I should have spelled this out more precisely. What I meant was that the `foldr` (which is difficult to read) is replaced entirely by taking the `minimum` over some appropriate list. Something along these lines:
```haskell
let commonIndentation = minimum $ (maxBound:) $ countIndentation <$> tail_
```